### ISSUE TITLE
Extension fails to load with GNOME 3.32

### ISSUE THREAD

**Issue by ronjouch** [pulb/mailnag-gnome-shell#57]
The usual! 😄, GNOME isn't exactly friendly with extensions backwards compat, eh 😕.

Here are the errors I see in `journalctl`:

```
mailnag[1380]: INFO (2019-03-16 12:06:02): Successfully enabled plugin 'soundplugin'.
mailnag[1380]: INFO (2019-03-16 12:06:02): Successfully enabled plugin 'dbusplugin'.
mailnag[1380]: INFO (2019-03-16 12:06:02): Checking 1 email account(s).
gnome-shell[1175]: JS WARNING: [/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/extension.js 302]: assignment to undeclared variable k
gnome-shell[1175]: JS WARNING: [/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/extension.js 302]: assignment to undeclared variable v
gnome-shell[1175]: JS WARNING: [/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/indicator.js 166]: reference to undefined property "Name"
mailnag[1380]: INFO (2019-03-16 12:07:04): Checking 1 email account(s).
gnome-shell[1175]: JS ERROR: Exception in callback for signal: MailsAdded: TypeError: this._construct is not a function
                   newClass@resource:///org/gnome/gjs/modules/_legacy.js:114:32
                   _addMailItems@/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/indicator.js:310:15
                   wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                   _updateMenu@/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/indicator.js:237:5
                   wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                   setMails@/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/indicator.js:421:3
                   wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                   _handle_new_mails@/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/extension.js:135:4
                   wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                   _onMailsAdded@/usr/share/gnome-shell/extensions/mailnag@pulb.github.com/extension.js:111:3
                   wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                   _emit@resource:///org/gnome/gjs/modules/signals.js:142:27
                   _convertToNativeSignal@resource:///org/gnome/gjs/modules/overrides/Gio.js:169:5
```

I'm using the official [Arch packages](https://www.archlinux.org/packages/?arch=any&arch=x86_64&q=mailnag), so currently:

- mailnag 1.2.1-3
- mailnag-gnome-shell 3.30.0-1

**Comment by pulb** [comment]
Thanks for reporting. This has been fixed in commit 7a3cbd0dbdc6ce5132f69771c2ae3bc2ed561713. The blue mail-count label still won't show up for some reason though.

**Comment by ronjouch** [comment]
> Thanks for reporting. This has been fixed in commit [7a3cbd0](https://github.com/pulb/mailnag-gnome-shell/commit/7a3cbd0dbdc6ce5132f69771c2ae3bc2ed561713).

@pulb 👍, thanks! 

> The blue mail-count label still won't show up for some reason though.

Oh, sad, that's the main reason I use mailnag (contrarily to in-your-face notifications, mailnag doesn't interrupt/distract, but information is still visible anytime I peek at the discrete the blue mail-count label). What's mailnag's utility, then? Just the sound? 😕

**Comment by pulb** [comment]
@ronjouch don't worry, I'm going to figure out what's wrong with that mail counter :wink:

**Comment by pulb** [comment]
All issues fixed in [v3.32](https://github.com/pulb/mailnag-gnome-shell/releases/tag/v3.32.0).
Should also be available on extensions.gnome.org soon.

**Comment by ronjouch** [comment]
> All issues fixed in [v3.32](https://github.com/pulb/mailnag-gnome-shell/releases/tag/v3.32.0).
> Should also be available on extensions.gnome.org soon.

@pulb awesome! As always, thanks for the continued maintenance and responsiveness 👍👍👍.

One piece of good news, by the way: your release notes say *"Port to ES6 (NOTE: this release won't work in GNOME < 3.32)"*. I think that's incorrect. If I understand correctly https://github.com/maoschanz/emoji-selector-for-gnome/issues/63#ref-commit-76b3644 , support for ES6 classes has been there since 3.26. It's mandatory now in 3.32, but you don't have to restrict the new version to 3.32, requiring ≥ 3.26 should be enough.

**Comment by pulb** [comment]
@ronjouch: Thanks, good to know, I corrected the release notes.
